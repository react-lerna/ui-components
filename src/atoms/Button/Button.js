import React, { Component } from 'react'
import styled, { css } from 'styled-components'
import { width, color, space } from 'styled-system'

export const StyledButton = styled.button`
  ${width};
  ${color};
  ${space};
  display: block;
  border-radius: 6px;
  border: 1px solid ${props => (props.secondary ? props.theme.colors.secondary.main : props.theme.colors.primary.main)};
  padding: 10px;
  color: ${props => !props.color && props.theme.colors.grey.main};
  background-color: ${props => (props.secondary ? props.theme.colors.secondary.bg : props.theme.colors.primary.bg)};
  ${props =>
  !props.disabled &&
  css`
  &:hover {
    background-color: ${props => (props.secondary ? props.theme.colors.secondary.bgAlt : props.theme.colors.primary.bgAlt)};
  },
  `}
  ${props =>
  props.disabled &&
  css`
      background-color: ${props => props.theme.colors.grey.light};
      border-color: ${props => props.theme.colors.grey.main};
    `}
`

class Button extends Component {
  render() {
    return <StyledButton {...this.props}>{this.props.children}</StyledButton>
  }
}

Button.defaultProp = {
  secondary: false,
}

export default Button
