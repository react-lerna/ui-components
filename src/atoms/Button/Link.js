import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export const StyledLink = styled.a`
  display: block;
  color: ${props => !props.color && props.theme.colors.secondary.dark};
`

class Link extends Component {
  render() {
    const { label, url, isExternal } = this.props
    return (
      <StyledLink {...this.props} target={isExternal && '_blank'} href={url}>
        {label}
      </StyledLink>
    )
  }
}

Link.propTypes = {
  label: PropTypes.string,
  secondary: PropTypes.bool,
  url: PropTypes.string,
  isExternal: PropTypes.bool,
}

Link.defaultProps = {
  secondary: false,
  url: '#',
}

export default Link
