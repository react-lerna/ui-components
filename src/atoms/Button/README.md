# Button
The Button component has the following options:

<br/>

## Exemple

```js
{
    label: 'Button text',
    color: 'primary.dark',
    secondary: true,
    disabled: false
}
```

<br/>
<br/>

## Types

```js
{
    label: string,
    color: string,
    secondary: boolean,
    disabled: boolean
}
```

<br/>
<br/>

### label

use any string like:

```js
label: 'anything text'
```

<br/>

### color

```code
use a string that is:

1. option from the design system
    - 'primary.main'
    - 'primary.dark'
    - 'secondary.dark'
    - ...

2. color name
    - 'red'
    - 'blue'

3. color code
    - 'rgb(12,66,120)'
    - 'hsl(219, 100%, 35%)'
    - '#333fff'
```

<br/>

### secondary

use true or false (default is false)

```js
secondary: true
```

<br/>

### disabled

use true or false (default is false)

```js
disabled: false
```
