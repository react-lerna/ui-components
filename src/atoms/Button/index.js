import {Component} from 'react'

class Button extends Component {
  render() {
    return <h1>Hello, {this.props.name}</h1>;
  }
}
export default Button
