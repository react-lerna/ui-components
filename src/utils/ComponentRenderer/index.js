export { default as ComponentRenderer } from './ComponentRenderer'
