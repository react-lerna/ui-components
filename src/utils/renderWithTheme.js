import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

import theme from '../theme'

export function renderWithTheme(component) {
  return renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
