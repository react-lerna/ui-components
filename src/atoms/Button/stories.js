import React, { Fragment } from 'react'
import { storiesOf } from '@storybook/react'
import Button from '.'

const mock1 = {
  title: 'Button',
  fontSize: [2, 3, 4],
}
const mock2 = {
  title: 'Secondary Button',
  secondary: true,
}
const mock3 = {
  title: 'Disabled Button',
  disabled: true,
}
const mock4 = {
  title: 'Primary Button with white text color',
  color: '#fff',
}
const mock5 = {
  title: 'Secondary Button with primary text color',
  secondary: true,
  color: 'primary.dark',
}

storiesOf('Atoms|Button', module)
  .add('Primary', () => <Button {...mock1} />)
  .add('Secondary', () => <Button {...mock2} />)
  .add('Disabled', () => <Button {...mock3} />)
  .add('Mix', () => (
    <Fragment>
      <Button {...mock4} />
      <Button {...mock5} />
    </Fragment>
  ))
