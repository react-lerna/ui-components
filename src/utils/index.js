import { ComponentRenderer } from './ComponentRenderer'

export default {
  ComponentRenderer,
}
