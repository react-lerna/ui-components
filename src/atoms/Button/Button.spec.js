import React from 'react'
import 'jest-styled-components'

import { renderWithTheme } from '../../utils/renderWithTheme'
import Button from './Button'

const mock = {
  label: 'Secondary Button with primary text color',
  secondary: true,
  color: 'primary.dark',
}

test('Button display correctly', () => {
  let tree = renderWithTheme(<Button {...mock} />).toJSON()
  expect(tree).toMatchSnapshot()
})
