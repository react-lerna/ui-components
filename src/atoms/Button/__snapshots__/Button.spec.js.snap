// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button display correctly 1`] = `
.c0 {
  color: hsl(80,50%,35%);
  display: block;
  border-radius: 6px;
  border: 1px solid hsl(190,100%,65%);
  padding: 10px;
  background-color: hsl(190,50%,85%);
}

.c0:hover {
  background-color: hsl(190,50%,92%);
}

<button
  className="c0"
  color="primary.dark"
  title="Secondary Button with primary text color"
>
  Secondary Button with primary text color
</button>
`;
